### PR TITLE
feat: Set up algebra-plugin repository

### PR DESCRIPTION
**Pull request by niermann999**
This PR puts the basic LA container functionality for detray into the algebra-plugin repo and adds the plugin unittest from detray.

### CODE REVIEW

File: tests/common/include/tests/common/test_plugin.inl [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+#include <cmath>
+#include <climits>
+
+#include <gtest/gtest.h>
+
+/// @note __plugin has to be defined with a preprocessor command
+using namespace algebra;
+
+// Two-dimensional definitions
+using point2cart = __plugin::cartesian2::point2;
```
Comment (asalzburger) [inline tests/common/include/tests/common/test_plugin.inl:19]: I think you branched off a bit too early, in a newer version of `detray` I have removed the strongly typed attempt that makes a difference between `cartesian::point2` and `polar::point2`.

I moved all `point3, vector3, point2, vector2` out into the `__plugin` namespace, this makes the code way simpler as you don't have to deduce the type from the local representation nor the `transform_type`.

I also removed the transform templating quite a bit.

**Review by asalzburger — CHANGES_REQUESTED** [review]
There's a few comments that need addressing.

File: common/include/algebra/common/types.hpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    using array_t = std::array<value_type, kDIM>;
+
+    template <typename value_type>
+    using vector_t = std::vector<value_type>;
+
+    template <typename key_type, typename value_type>
+    using map_t = std::map<key_type, value_type>;
+
+    template< class... types>
+    using tuple_t = std::tuple<types ...>;
```
[the remark this answers is not in the document]
Comment (niermann999) [reply common/include/algebra/common/types.hpp:27]: Yes, I think to figure this out is the main point right now. I added the Vc types here, too

File: external/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+
```
[the remark this answers is not in the document]
Comment (niermann999) [reply external/CMakeLists.txt:1]: This belongs more to the next pull request and is actually in preparation of the addition of Vc/veccore.

File: tests/common/include/tests/common/test_plugin.inl (hunk 1 above)
Comment (niermann999) [reply tests/common/include/tests/common/test_plugin.inl:19]: Ah, damn. I will have to rework parts of the detray integration then.

File: common/include/algebra/common/types.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulgessinger) [reply common/include/algebra/common/types.hpp:27]: Also, maybe we'd want to default to `std::unordered_map` or is ordering relevant?

[the remark this answers is not in the document]
Comment (asalzburger) [reply common/include/algebra/common/types.hpp:27]: We may need to use both for detray.

**Review by asalzburger — APPROVED** [review]

**Pull request merged**